### ISSUE TITLE
Question: Showing initial slides before JS is executed (or if it's disabled)

### ISSUE THREAD

**Issue by koobazaur** [woocommerce/FlexSlider#1627]
Hey folks!

I am working on a Wordpress site and its theme uses Flexslider on the homepage. It's negatively affecting the ratings on Google Page Speed Insights since it delays above-the-fold content from showing up. I noticed the slides don't actually show up until the flexcode JS code is initiated, after everything else has been loaded (in jQuery document ready).

I would rather not have to re-code our theme to use a different slider, so I am wondering if there is a way to at least show the first few slide images instantaneously without waiting for flex to be initted or in case JS is disabled?

For reference, here is the website so you can see it yourself: https://cdanews.com/ . 

Please advise.

Best,
Jakub

**Comment by TSS-IEB** [comment]
Timely.  I too have this query and was wondering of the easy fix was to just make the script asynchronous (e.g; <script async src="my.js">).
In Flexslider's case this would be:
<script async src="https://ajax.googleapis.com/ajax/libs/jquery/1.6.2/jquery.min.js"></script>
<script async src="../sc_code/flexslider/jquery.flexslider-min.js"></script>
Granted this wont display the first few slides instantaneously, but in the real world would it help page load speed?
Any comments?

**Comment by koobazaur** [comment]
I already set mine to async but the problem is that we have several sliders above the fold so nothing shows up until after page load. Not only does it look weird, but Google is dinging us for nothing showing above-the-fold content quickly enough (its just big blank space with slider titles)

**Comment by cseto** [comment]
@koobazaur Did you find a solution to the problem?

**Comment by fry2k** [comment]
I had the same problem a few months ago with google pagespeed and the above the fold content of our flexslider.
I solved this with adding a .unfinished css class to the ul-tag of the slides which is removed after initializing the flexslider. I wrote some custom rules based on the unfinished class and manipulate the slider html markup like I needed it. I spend a lot of time and now it works like expected and pagespeed is fine.

Would be great if we could have an easier implementation inside flexslider but I don't think that this is realistic.

I have also some other pagespeed hacks like choosing the right img src based on the device, so don't scare what we did there, just for reference http://www.buecher.de
